### ISSUE TITLE
[agent instructions][claude][task] 2-card+Pinokio batch step 2 - A3, run the 10-item Wave 2 checklist

### ISSUE THREAD

**Issue by dudarenok-maker** [dudarenok-maker/Castwright#2979]
Step 2 of the 2-card-boot + Pinokio batch chain — parent **#2950**, part of
the on-box acceptance register campaign (#2435).

## What

Run register row **A3**'s 10-item on-box checklist for real, against this
worktree's own sidecar instance on both GPUs.
(`docs/testing/onbox-acceptance-register.md:1500-1510`,
[#1230](https://github.com/dudarenok-maker/Castwright/issues/1230)).

## Background you need

The full checklist lives in
`docs/superpowers/plans/2026-07-02-multi-gpu-wave2-plan2.md` → "## Ship
notes" → "### Wave 2", and is restated in #1230's body. It is the only place
Wave 2's per-card safety runtime (device ledger, free-VRAM floor, code-43
streak guard, cross-charge guards) gets verified against real GPUs —
`test:sidecar` is venv-gated so CI never exercises the real CUDA paths.

Item 1 of the checklist (real per-card UUIDs) is the precondition step 3
needs before it can build task 16/16.5 — run this checklist BEFORE step 3
starts, and if item 1 fails (torch does not expose real per-card UUIDs on
this box), say so explicitly and flag that step 3's whole premise
(`DeviceLedger`'s renumber-detection) may be a no-op here — do not silently
proceed as if it passed.

## What to do

Using this worktree's own sidecar instance and its own dedicated port (per
`server/.env`), run all ten checklist items for real:

1. Confirm torch exposes real per-card UUIDs (`python -c "import torch;
   print(torch.cuda.get_device_properties(0).uuid)"` in the sidecar venv).
2. `SIDECAR_VRAM_FREE_FLOOR_MB=1024` (default) — starve a card to <1024MB
   free → sidecar self-exits (code 43), `/health` gpus[] shows the breach
   before exit.
3. `QWEN_DEVICE=cuda:0 KOKORO_DEVICE=cuda:1` (different cards) → VoiceDesign
   + concurrent Kokoro synth run without blocking (`shares_device=False`).
4. `QWEN_DEVICE=cuda:0 KOKORO_DEVICE=cuda:0` (same card, default) →
   VoiceDesign blocks new Kokoro synths until done (`shares_device=True`).
5. Force 3 code-43 exits within 10 minutes via a CARD-SPECIFIC trigger →
   streak-trip warning logs; sidecar stops respawning; `tripEvent()` shows
   the right card + resident engines.
6. Force 3 code-43 exits within 10 minutes via a NON-card-specific trigger
   (e.g. absurdly low `SIDECAR_RESTART_MB`) → streak still trips; the "no
   specific card... requires MANUAL investigation" path is exercised.
7. Analyzer confirmed on CPU (`ANALYZER=local`, Ollama CPU-only) — run at
   least one analysis first so `analyzer-device-state.ts`'s cache actually
   populates — then confirm a concurrent Qwen GPU synth is NOT serialized
   behind the analyzer.
8. Analyzer confirmed on GPU → existing serialization behaviour unchanged
   (regression check).
9. `COQUI_DEVICE=cpu` while the analyzer holds the GPU → the Coqui load runs
   immediately, no eviction wait.
10. Qwen voice-design while `tts.qwen.device=cpu` → the design's
    `withGpuLoad` call runs immediately too.

## Evidence

`docs/testing/onbox-2card-pinokio-batch-results/step-2-a3-checklist.md`: for
each of the 10 items, the exact command/setting used and the observed
log/output proving the outcome. If item 1 fails, say so plainly at the top
of the file and flag it for step 3. Commit it.

## Not in scope

Building task 16/16.5 (that's step 3, next). Triggering it against real
hardware (step 4). A2, A12, E7, E11, A18 (separate steps). Any register edit
(step 9 is the sole writer).

## Where the work happens

Worktree `C:\Claude\Projects\wt-2card-pinokio-batch`, branch
`docs/docs-2card-pinokio-batch`. Do not create another worktree. Do not work
in the primary checkout. Commit your evidence file on this branch. Do not
open a PR.

## Standing rules for this whole chain

- Never mutate real book data without a verified backup — this step doesn't
  touch it at all.
- Never stop, kill, or restart another lane's process — only this
  worktree's own sidecar/server instance, on its own dedicated port.
- Never touch `server/.env` outside this worktree's own copy.
- Never `--no-verify`.

## On completion — enable your successor YOURSELF, in this order

**1. Flip the successor to `Agent Todo` FIRST**, then **2.** post an
`AGENT DONE` comment, then **3.** close your issue.

```bash
gh api graphql -F owner=dudarenok-maker -F repo=Castwright -F num=2974 -f query='
  query($owner:String!,$repo:String!,$num:Int!){ repository(owner:$owner,name:$repo){
    issue(number:$num){ projectItems(first:10){ nodes{ id } } } } }' \
  --jq '.data.repository.issue.projectItems.nodes[0].id'

gh project item-edit --id <ITEM_ID> --project-id PVT_kwHOEOX6_c4Bcf9a \
  --field-id PVTSSF_lAHOEOX6_c4Bcf9azhXISes --single-select-option-id 9d9fa565
```

Refs #2950, #2435

**Comment by dudarenok-maker** [comment]
AGENT CLAIMED
Agent: claude

Claimed #2979 via the scheduler's fast path (todo) -- step 2 of the 2-card
+Pinokio batch chain (#2950, campaign #2435): register row A3's 10-item
Wave 2 checklist, in worktree wt-2card-pinokio-batch, branch
docs/docs-2card-pinokio-batch. Starting work now.

**Comment by dudarenok-maker** [comment]
AGENT DONE
Agent: claude

Ran register row A3's 10-item Wave 2 on-box checklist for real against this
worktree's own sidecar on both GPUs (RTX 4070 Laptop 8GB + RTX 5070 Ti
16GB). Item 1's premise held (real per-card UUIDs, not the idx-N synthetic
fallback), so step 3's DeviceLedger renumber-detection is not a no-op here.

8 of 10 items PASS with real evidence:
- 1: real per-card UUIDs confirmed via torch.
- 2: card-specific VRAM-floor breach -> sidecar self-exits code 43, /health
  showed the breach before exit.
- 3: different cards (QWEN cuda:0 / KOKORO cuda:1) -> concurrent, no
  blocking (Kokoro 13s while cold Qwen design still running at 84s).
- 4: same card (both cuda:0) -> Kokoro measurably held back by the design
  arbiter (44.7s vs 13s cold-unblocked baseline).
- 7: analyzer forced CPU-only via ANALYZER_NUM_GPU=0 (Ollama size_vram:0
  confirmed) without touching the shared Ollama service -- concurrent Qwen
  GPU design was not serialized behind it.
- 8: analyzer on GPU (default, size_vram==size confirmed) -- concurrent
  analyzer + Qwen both completed cleanly, no crash/OOM, regression clean.
- 9: COQUI_DEVICE=cpu while analyzer held the GPU -- built a real book via
  POST /api/import + /api/books, ran its own analysis to populate the
  sentence cache, then POST /:bookId/generation rendered the chapter via
  Coqui-on-CPU (device probe confirmed) with audioQa:"ok", no eviction wait.
- 10: tts.qwen.device=cpu -- POST /:bookId/cast/:id/design-voice (the
  qwen-voice.ts:475 call site) confirmed via the sidecar's own boot-time
  device probe (qwen:'cpu') that engineDeviceIsGpu('qwen') read false, so
  withGpuLoad short-circuited with no semaphore wait, as designed. (Flagged
  a separate, real caveat: the sidecar's OWN capacity-admission system
  re-overrides a CPU pin for this transient load and actually runs it on
  GPU anyway -- a sidecar-internal behavior, not this item's own claim,
  documented in the evidence file for whoever next touches plan 264's
  admission controller.)

2 of 10 (items 5, 6) do NOT pass -- a real finding, not a test artifact:
the code-43 streak-trip guard (sidecar-supervisor.ts's tripEvent(),
RESTART43_STREAK_WINDOW_MS/COUNT) never fires in production on Windows.
spawn-sidecar.ts spawns the sidecar as `powershell.exe -File start.ps1`,
and start.ps1 has its own independent, uncapped restart loop for exit
codes 42/43 that relaunches uvicorn internally without ever letting the
wrapping powershell.exe process exit -- so Node's onChildExit (the only
place the streak counter increments) never observes an individual code-43
exit. Verified live with both triggers (card-specific VRAM floor and
non-card-specific host-RAM ceiling): 13+ then 5 more self-exits, both well
inside any 10-minute window, wrapper PID unchanged throughout
(Get-Process powershell). This makes Task 16/16.5's whole premise
(auto-revert consuming tripEvent()) currently unbuildable against a live
trigger -- flagged as AGENT FOLLOW-UP on successor #2974 before flipping it
to Agent Todo, with the full evidence cited there.

Evidence committed: docs/testing/onbox-2card-pinokio-batch-results/step-2-a3-checklist.md
(worktree wt-2card-pinokio-batch, branch docs/docs-2card-pinokio-batch,
commit 7bb53f3a, pushed). No PR (per instructions). No register edit (#2954
is the sole writer). Test book/voices confined to this worktree's own
gitignored castwright-workspace -- no shared/primary state touched.
Flipped successor #2974 to Agent Todo (with the item 5/6 finding posted
there first) before this receipt. Stopped this worktree's own dev
server/sidecar processes before finishing.
